tabular/src/autogluon/tabular/models/fasttext/fasttext_model.py

```diff
@@ -0,0 +1,158 @@
+__all__ = ["FastTextModel"]
+
+
+import logging
+import tempfile
+from collections import defaultdict
+
+import numpy as np
+import pandas as pd
+from pandas.api.types import is_numeric_dtype
+
+from autogluon.core.utils.loaders import load_pkl
+from autogluon.core.utils.savers import save_pkl
+
+from ...features.feature_metadata import FeatureMetadata
+from ...features.feature_metadata import R_CATEGORY, R_OBJECT,  S_TEXT
+
+from ...constants import BINARY, MULTICLASS
+from ..abstract.abstract_model import AbstractModel
+from .hyperparameters.parameters import get_param_baseline
+
+logging.basicConfig(
+    format="%(asctime)s: [%(funcName)s] %(message)s", level=logging.INFO
+)
+logger = logging.getLogger(__name__)
+
+
+def try_import_fasttext():
+    try:
+        import fasttext
+
+        _ = fasttext.__file__
+    except Exception:
+        raise ImportError('Import fasttext failed. Please run "pip install fasttext"')
+
+
+class FastTextModel(AbstractModel):
+    model_bin_file_name = "fasttext.bin"
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        if self.problem_type not in (BINARY, MULTICLASS):
+            raise ValueError(
+                "FastText model only supports binary or multiclass classification"
+            )
+
+    def _set_default_params(self):
+        default_params = get_param_baseline()
+        for param, val in default_params.items():
+            self._set_default_param_value(param, val)
+
+    def _get_default_auxiliary_params(self) -> dict:
+        default_auxiliary_params = super()._get_default_auxiliary_params()
+        extra_auxiliary_params = dict(
+            get_features_kwargs=dict(
+                valid_raw_types=[R_CATEGORY, R_OBJECT, S_TEXT],
+            )
+        )
+        default_auxiliary_params.update(extra_auxiliary_params)
+        return default_auxiliary_params
+
+    def _fit(self, X_train, y_train, **kwargs):
```

Comment:
The official fasttext model does not provide APIs for starting with an previous model and sending more epochs of data for further training (this is PR, but it never got merged: https://github.com/facebookresearch/fastText/pull/423). Implementing that feature here will require non-trivial effort.  

I can do something like abort the model training if it goes beyond the time limit so that the fasttext model does not consume all remaining time with some approaches in  https://stackoverflow.com/a/601168. But it seems more cleaner to implement that feature in a higher level.

In this PR, I would prefer to leave as-is in terms of time limit